pkg/kv/kvserver/kvflowcontrol/replica_rac2/processor.go

```diff
@@ -582,21 +525,10 @@ func (p *processorImpl) SetEnabledWhenLeaderRaftMuLocked(
 		return
 	}
 	log.VEventf(ctx, 1, "enabled v2 protocol using v1 priority encoding")
+	p.leaderID = raftBasicState.Leader
```

Comment:
Should this be updated somewhere else, together with all other things that we need to update when `leaderID` changes? Placing this write here potentially (temporarily?) breaks invariants in `p`.